web-jehlomat/src/screens/SetNewPassword/SetNewPassword.tsx

```diff
@@ -0,0 +1,144 @@
+﻿import React, {FunctionComponent} from "react";
+import {useHistory} from "react-router-dom";
+import {useMediaQuery} from "@mui/material";
+import {Formik, FormikHelpers} from "formik";
+import {AxiosResponse} from "axios";
+import * as yup from "yup";
+import API from "config/baseURL";
+import {media} from "utils/media";
+import hasOwnProperty from "utils/hasOwnProperty";
+import SignInImage from "assets/images/sign-in.svg";
+import {Header} from "Components/Header/Header";
+import PrimaryButton from "Components/Buttons/PrimaryButton/PrimaryButton";
+import TwoColumns from "Components/Layout/TwoColumns";
+import {Input,Inputs,FormTitle,TransparentForm,Buttons,Illustration} from "./Components";
+
+const texts = {
+  TITLE: "Nové heslo",
+  MOBILE_TITLE: "Nové heslo",
+  INPUT__PASS__LABEL: "Nové heslo",
+  INPUT__PASS__PLACEHOLDER: "Snadno zapomatovatelné, těžce uhodnutelné",
+  INPUT__RE_PASS__LABEL: "Kontrola hesla",
+  INPUT__RE_PASS__PLACEHOLDER: "Zadejte stejné heslo",
+  VALIDATIONS__REQUIRED: "Povinná položka",
+  VALIDATIONS__SYMBOLS: "Heslo musí obsahovat číslo, velké a malé písmeno",
+  VALIDATIONS__MIN_LENGTH: "Heslo musí být 8 znaků dlouhé",
+  VALIDATIONS__MATCH: "Hesla musí být stejná",
+  BUTTONS__SEND: "Nastav heslo",
+};
+
+interface ResetPasswordFormProps {
+  password: string;
+  rePassword: string;
+}
+
+const init: ResetPasswordFormProps = {
+  password: '',
+  rePassword: '',
+}
+
+const schema = yup.object({
+  password: yup.string()
+    .matches(/^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])/, texts.VALIDATIONS__SYMBOLS)
```

Comment:
`/^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])/` I would move it to a constant. I think, the sign-in logic also needs it

Reply:
I was thinking the same way, but I wasn't sure how are you about having refactoring within the feature branch (I would update occurrences in an organization and user registration).